### DISCUSSION TITLE
Reset gazebo world results in loss of control of HSR

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#74] [Q&A]
For the simulator, whenever I reset the gazebo world, the Python controllers will stop working. Does anyone have any fix to that? Thanks!

**Comment by masudam** [comment]
Talking about the "/gazebo/reset_simulator" service? If the Gazebo simulation time is reset, the ROS time does not roll back, causing a mismatch and leading to malfunction. As for "/gazebo/reset_world," it only resets the objects to their initial positions, so errors generally do not occur. (If you repeat the reset process multiple times, it is uncertain whether it will work accurately.)
About reset_world/simulator services : [URL](https://answers.gazebosim.org//question/24566/difference-between-reset_simulation-and-reset_world/)
